src/main/java/org/mitre/synthea/engine/State.java

```diff
@@ -1313,19 +1378,38 @@ public boolean process(Person person, long time) {
   public static class Procedure extends Delayable {
     private List<Code> codes;
     private String reason;
+    // For GMF 1.0 Support
     private RangeWithUnit<Long> duration;
     private String assignToAttribute;
     private Long stop;
+    // For GMF 2.0 Support
+    private Distribution distribution;
+    private String unit;
+
+    @Override
+    protected void initialize(Module module, String name, JsonObject definition) {
+      super.initialize(module, name, definition);
+      if (distribution != null && !distribution.validate()) {
+        throw new IllegalStateException(
+            String.format("State %s contains an invalid distribution", this.name));
+      }
+    }
 
     @Override
     public Procedure clone() {
       Procedure clone = (Procedure) super.clone();
+      clone.codes = codes;
```

Comment:
I think the super.clone will take care of copying all of the member variables so you can remove lines 1401-1406 inclusive.